### ISSUE TITLE
Apple HEIC support

### ISSUE THREAD

**Issue by Webbeh** [cowai/camerarawpreviews#22]
It's no secret that Apple made that new picture format available on certain iPhones, and that there is no support, right now, for this format into Nextcloud.

Would it be possible to build this in this plugin ?

You should check "Tifig" out, here on github : https://github.com/monostream/tifig
That library is able to convert HEIC to different formats. Even if it just adds support to server owners being able to install that library/package on their system at first, it's still an improvement.

**Comment by steiny2k** [comment]
I think the more recent builds of ImageMagick (which this app uses to provide the previews for RAW files) support also HEIF / HEIC (see https://github.com/ImageMagick/ImageMagick/issues/507). So it's not necessarily required to rely on another graphics converter such as tifig.
Based on my current understanding, this means, this app could easily be enhanced to also tackle files with the .heif extension, and call up ImageMagick to generate a preview.

Besides this I've created a PR to the main Nextcloud Server component to add HEIF / HEIC support (see https://github.com/nextcloud/server/pull/10526). I'll tidy it up the next days and hope to make it into the upstream Nextcloud release.

In case Nextcloud upstream does not want to include my PR, I can adapt it to integrate it into this app, or even create a new app to release it separately.

For OwnCloud there's already a patch which is the foundation on which I've built my PR. See https://github.com/owncloud/core/pull/30093
